src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -23412,5 +23412,58 @@ void M()
             Assert.Equal("(System.Int32, System.String, System.Int32)", model.GetTypeInfo(tuple).ConvertedType.ToTestDisplayString());
             Assert.Equal(ConversionKind.NoConversion, model.GetConversion(tuple).Kind);
         }
+
+        [Fact]
+        [WorkItem(20494, "https://github.com/dotnet/roslyn/issues/20494")]
+        public void MoreGenericTieBreaker_01()
+        {
+            var source =
+@"using System;
+public class C
+{
+    public static void Main()
+    {
+        A<A<int>> a = null;
+        M1(a); // ok, selects M1<T>(A<A<T>> a)
+
+        var b = default(ValueTuple<ValueTuple<int, int>, int>);
+        M2(b); // ok, should select M2<T>(ValueTuple<ValueTuple<T, int>, int> a)
+    }
+    public static void M1<T>(A<T> a) { Console.Write(1); }
+    public static void M1<T>(A<A<T>> a) { Console.Write(2); }
+
+    public static void M2<T>(ValueTuple<T, int> a) { Console.Write(3); }
+    public static void M2<T>(ValueTuple<ValueTuple<T, int>, int> a) { Console.Write(4); }
+}
+
+public class A<T> {}";
+            var comp = CompileAndVerify(source, additionalRefs: s_valueTupleRefs, expectedOutput: "24");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/20583")]
+        [WorkItem(20494, "https://github.com/dotnet/roslyn/issues/20494")]
+        [WorkItem(20583, "https://github.com/dotnet/roslyn/issues/20583")]
+        public void MoreGenericTieBreaker_02()
```

Comment:
No, it is intentional, and it is not a bug. Between the two overloads
``` c#
M<T, U, V>(((int, T), int, int, int, int, int, int, int, int, (int, U), V))
M<T, U, V>((T, int, int, int, int, int, int, int, int, U, (int, V)))
```
which could also be written
``` c#
M<T, U, V>(VT<VT<int, T>, int, int, int, int, int, int, VT<int, int, VT<int, U>, V>>)
M<T, U, V>(VT<T, int, int, int, int, int, int, VT<int, int, U, VT<int, V>>>)
```
when called on a value of type
``` c#
((int, int), int, int, int, int, int, int, int, int, (int, int), (int, int))
```
which could also be written
``` c#
VT<VT<int, int>, int, int, int, int, int, int, VT<int, int, VT<int, int>, VT<int, int>>
```
The former is more specific on the first type parameter of the argument, and they agree on type arguments 2-7, and in type argument position 8 we have a value of type
``` c#
VT<int, int, VT<int, int>, VT<int, int>
```
for which neither
``` c#
VT<int, int, VT<int, U>, V>
```
nor
``` c#
VT<int, int, U, VT<int, V>>
```
is more specific.

Since the former is more specific for type argument 1 of its first parameter, and the second is not more specific for any type argument of any parameter, the first overload is more specific.